### PR TITLE
Basic spatial data structures

### PR DESCRIPTION
**Pull request by zeroset**
This is a first draft for basic spatial datatypes and the corresponding python bindings. 
For LengthUnit Construction we  already have a problem (@schroedtert @JetteSchumann @Ozaq):
The Constructor takes the input unit as template parameter. Pybind can only bind to a concrete instantiation of the template. I have no idea how to pass the actual unit from python to C++. The only Idea i came up with is to create factory methods in python like: `CreateLengthUnitFromM(quantity), ...` . 
Any Idea?
Sorry, the PR is based on #38. That is why the Changed Files are so many.

### CODE REVIEW

File: cpp/libcore/source/geometry/coordinate.hpp [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+namespace jps
+{
+/// Spatial Coordinate
+///
+/// Represents a spatial coordinate in the world (geometry).
+class Coordinate
+{
+    LengthUnit m_x;
+    LengthUnit m_y;
+    Level m_lvl;
```
Comment (schroedtert) [inline cpp/libcore/source/geometry/coordinate.hpp:15]: Is level a number to identify the level or a data structure to hold all structures within a certain level?

File: cpp/libcore/source/geometry/special_area.hpp [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+#include "area.hpp"
+
+#include <cstdint>
+namespace jps
+{
+class SpecialArea
+{
+    Area m_area;
+    std::uint32_t m_id{};
```
Comment (schroedtert) [inline cpp/libcore/source/geometry/special_area.hpp:11]: What about using `Identifiable`?

File: cpp/libcore/source/geometry/coordinate.hpp (hunk 1 above)
Comment (zeroset) [reply cpp/libcore/source/geometry/coordinate.hpp:15]: There is a Level class already, but actually I don't know whats in there. Maybe just an small integral type?

File: cpp/libcore/source/geometry/special_area.hpp (hunk 2 above)
Comment (zeroset) [reply cpp/libcore/source/geometry/special_area.hpp:11]: We need some id which is the same as coming from the DXF file. I have no idea which int type is sufficient and we need internal identifiable.

File: cpp/libcore/source/geometry/coordinate.hpp (hunk 1 above)
Comment (JetteSchumann) [reply cpp/libcore/source/geometry/coordinate.hpp:15]: I am not sure if we need a Level Class... actually a Level is represented by an ID extracted from the geometry file where the user has to stick to naming patterns.

File: cpp/libcore/source/geometry/special_area.hpp (hunk 2 above)
Comment (JetteSchumann) [reply cpp/libcore/source/geometry/special_area.hpp:11]: btw: I just read that 24bit colors are only writable with QCAD professional. I will check the community edition

Comment (zeroset) [reply cpp/libcore/source/geometry/special_area.hpp:11]: Ok, maybe we should consider switching to libeCAD as reference CAD Editor.

File: cpp/libcore/source/geometry/coordinate.hpp (hunk 1 above)
Comment (zeroset) [reply cpp/libcore/source/geometry/coordinate.hpp:15]: I would prefer having named types for things like this. This abstraction comes for free at least to some extend :D.

**Review by schroedtert — APPROVED** [review]

File: cpp/libcore/source/geometry/coordinate.hpp [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "geometry/length_unit.hpp"
+#include "geometry/level.hpp"
+
+namespace jps
+{
+/// Spatial Coordinate
+///
+/// Represents a spatial coordinate in the world (geometry).
+class Coordinate
```
Comment (schroedtert) [inline cpp/libcore/source/geometry/coordinate.hpp:11]: Do we need a class or is struct sufficient?

**Pull request merged**